src/push_context.rs

```diff
@@ -428,6 +404,29 @@ fn determine_names(
         })
     };
 
+    // If a flake name is explicitly provided, validate that name, otherwise use the
+    // inferred repository name
+    let upload_name = if let Some(name) = explicitly_provided_name {
+        let num_slashes = name.matches('/').count();
+
+        if num_slashes == 0
+            || !name.is_ascii()
+            || name.contains(char::is_whitespace)
+            || num_slashes > 1
+        {
+            let error_msg = if subgroup_renaming_explicitly_disabled {
+                "The argument `--name` must be in the format of `owner-name/repo-name` and cannot contain whitespace or other special characters"
+            } else {
+                "The argument `--name` must be in the format of `owner-name/subgroup/repo-name` and cannot contain whitespace or other special characters"
```

Comment:
This message seems wrong -- the argument `--name` can be in _either_ `owner/repo` or `owner/sub/repo` formats.

Reply:
Updated this to be more descriptive

Reply:
Actually, no, I updated it again to provide a better error message for explicitly provided upload names (which always must have one slash).